sdk/src/main/java/io/dapr/client/BaseSubscribeConfigHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.client;
+
+import io.dapr.client.domain.SubscribeConfigurationResponse;
+import java.util.logging.Logger;
+
+/**
+ * Base class to handle configuration subscribe response.
+ * Application needs to extend this calls to have better control on response handelling
+ */
+public class BaseSubscribeConfigHandler {
```

Comment:
Move the logic to DaprRuntime so we avoid spreading the number of singletons in the SDK.

You might just need to add the following to DaprRuntime:
```java
  private final Map<String, BiConsumer<String, String>> configurationChangeHandlers = Collections.synchronizedMap(new HashMap<>());

  public void registerConfigurationChangeHandler(String store, BiConsumer<String, String> handler) {
    this.configurationChangeHandlers.put(store, handler);
  }

  public void handleConfigurationChange(String store, String key, String value) {
    BiConsumer<String, String> handler = this.configurationChangeHandlers.get(store);
    handler.accept(key, value);
  }
```

Reply:
Did this change. The catch is `DaprRuntime.java` has default modifier so, we can not access the `registerConfigurationChangeHandler ` from examples.

I made `DaprRuntime` to 'public final` hope this should be fine.

Reply:
Sorry, this side effect was unexpected and undesired. Let me find a solution for this right now to merge this for 1.9.

Reply:
Let's remove it and keep the SpringBoot example without this layer of convenience. I have explored a few options and I think we will need further design discussions. So, I will make the changes now to simplify it until we create clarity for that.